### ISSUE TITLE
Make builds fully reproducible

### ISSUE THREAD

**Issue by aneeshusa** [servo/servo#11980]
This is a metabug, since this will require multiple PRs. Reproducible builds make it possible to independently verify the output of a particular build, strengthening the security of our release pipeline by enabling outside auditors. Since making builds reproducible also means controlling all of the inputs, this also makes it much easier to check out arbitrary revisions and successfully build them.

A good place to start for anyone interested in helping with this is looking for one of the sources of non-determinism listed under "Get deterministic builds" at the [reproducible-builds.org website](https://reproducible-builds.org/docs/).

**Comment by edunham** [comment]
Another place that a new contributor could jump in on this bug is locally building the same checkout of Servo twice, noting the locations of any files that differ between the two builds. For extra credit, track down the areas of the source involved in producing the differing files and identify the specific code causing the non-determinism :)

**Comment by aneeshusa** [comment]
Note that I tried to do what @edunham suggests against two `target/` dirs with the [diffoscope](https://diffoscope.org/) tool, but diffoscope was unable to complete the comparision - I don't remember if I ran out of memory or it just never returned. You will likely need to compare bits and pieces at a time.

**Comment by Coder206** [comment]
I am interested in helping out with this.

**Comment by aneeshusa** [comment]
@Coder206 this is a fairly large task and can't (shouldn't) be solved in one PR, so this is a metabug. I'd suggest looking at https://reproducible-builds.org/docs/ for a smaller piece of non-determinism to remove from our build and opening a separate issue/PR for it that links to this one to start.

**Comment by Coder206** [comment]
@aneeshusa So if I read correctly, you want me to find something that will suit me from the documentation, work on that and then PR my code?

**Comment by aneeshusa** [comment]
Right! I would also ask that you once you find something to work on in our build system, that you open an issue for it so we can assign it to you and as a place for any questions.

**Comment by jdm** [comment]
I'm removing the `E-less easy` tag since it's not clear that the earlier descriptions are really enough for someone to start working on this.

**Comment by nox** [comment]
Cc @metajack who is doing stuff in this area.
